config-svc/pkg/api/routes_v1.go

```diff
@@ -15,93 +15,137 @@
 package api
 
 import (
-	"encoding/json"
-	"errors"
 	"fmt"
-	"io/fs"
+	"io"
 	"net/http"
 	"os"
 
-	"github.com/kennygrant/sanitize"
+	"github.com/couchbase/tools-common/cbvalue"
+	"github.com/couchbaselabs/observability/config-svc/pkg/couchbase"
+	"github.com/couchbaselabs/observability/config-svc/pkg/prometheus"
+	"gopkg.in/yaml.v3"
 
 	v1 "github.com/couchbaselabs/observability/config-svc/pkg/api/v1"
 	"github.com/labstack/echo/v4"
 )
 
-const (
-	prometheusTargetsPathDevelopment = "./targets/%s.json"
-	prometheusTargetsPathProduction  = "/etc/prometheus/couchbase/custom/%s.json"
-)
-
-type prometheusTargets []struct {
-	Targets []string          `json:"targets"`
-	Labels  map[string]string `json:"labels"`
-}
+const defaultPrometheusConfigPath = "/etc/prometheus/prometheus.yml"
 
-func (s *Server) PostAddPrometheusTarget(ctx echo.Context) error {
-	var data v1.PostAddPrometheusTargetJSONBody
+func (s *Server) PostClustersAdd(ctx echo.Context) error {
+	var data v1.PostClustersAddJSONRequestBody
 	if err := ctx.Bind(&data); err != nil {
 		return err
 	}
 
-	labels := data.Labels.AdditionalProperties
-	if labels == nil {
-		labels = make(map[string]string)
+	secure := false
+	if data.CouchbaseConfig.Secure != nil && *data.CouchbaseConfig.Secure {
```

Comment:
it feels like this could be a one liner but I think that's my C days rearing their ugly head!

Reply:
Sadly no - if this were C I'd write `data.CouchbaseConfig.Secure != nil ? *data.CouchbaseConfig.Secure : false` but no ternaries in Go :-(

Reply:
Yeah because they always take ages to parse in your head! Defaulting would be much better but I don't think you can do it for generic YAML.

Reply:
Sadly that struct is auto-generated by the OpenAPI generator, so we'd have to override its templates. Not a bad idea, but likely more trouble than it's worth for the time being.

Reply:
Yeah I'd probably just go with a helper method on the struct type, e.g. 
```
(struct* data) IsSecure() bool {
  if data.CouchbaseConfig.Secure != nil && *data.CouchbaseConfig.Secure {
    return false
  }
  return true
}
```
Not sure if that is possible but simplifies it all here and you could do it for some of the others then.

Reply:
The definition of the data is

```go
type Cluster struct {
	CouchbaseConfig struct {
		ManagementPort *float32 `json:"managementPort,omitempty"`
		Password       string   `json:"password"`
		UseTLS         *bool    `json:"useTLS,omitempty"`
		Username       string   `json:"username"`
	} `json:"couchbaseConfig"`
	// ...
}
```

CouchbaseConfig is an inline struct so we can't define the method there - we could define it on `Cluster`, but that seems like a bit of a smell to me

Reply:
Normally we have helper methods elsewhere for this kind of stuff - there's some lovely mangling as you know required for some of the REST APIs. Overkill for just one thing but if we start having others it is probably worth it to simplify the methods.

Reply:
If I controlled that code I'd probably use something like https://github.com/guregu/null to handle the nulls... sadly no such luck here